src/Interpreters/InterpreterGrantQuery.cpp

```diff
@@ -28,6 +28,14 @@ namespace
         const ASTGrantQuery & query,
         const std::vector<UUID> & roles_to_grant_or_revoke)
     {
+        if (query.is_replace && !query.is_revoke)
+        {
+            if (roles_to_grant_or_revoke.empty())
```

Comment:
There can be also command `REPLACE GRANT NONE TO user1` (set granted roles for user1 to none), which should also work. That's why I suggested two boolean variables `query.replace_access` and `query.replace_granted_roles` instead one `query.is_replace`.

Reply:
@vitlibar Is it that `REPLACE GRANT NONE ON *.* TO user1` means `REVOKE ALL ON *.* FROM user` to revoke all privileges from user1 and keep it's roles unchanged.  
  
And `REPLACE GRANT NONE TO user1` means `REVOKE ALL FROM user`, just revoke all roles from it, and keep it's privileges unchanged.

Reply:
yes

Reply:
Can you please add those two `REPLACE GRANT NONE` cases to your test?

Reply:
@vitlibar I've submitted, PTAL